packages/shared/src/hooks/useOnPostClick.ts

```diff
@@ -30,13 +38,52 @@ export type FeedPostClick = ({
   optional?: PostClickOptionalProps;
 }) => Promise<void>;
 
+const getFeedQueryKeys = (client: QueryClient): QueryKey[] => {
+  const queryKeys = client
+    .getQueryCache()
+    .findAll()
+    .map((query) => query.queryKey);
+
+  // filter only query keys for supported feed layout v1 feeds
+  const keys = queryKeys.filter(
```

Comment:
Since this can happen many times while user clicks on articles `reduce` could help to do filter + map in the same iteration.

Reply:
Yeah, sure, non blocking.